core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -207,13 +207,17 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
   private var currentConfig = kafkaConfig
   private val dynamicConfigPasswordEncoder = maybeCreatePasswordEncoder(kafkaConfig.passwordEncoderSecret)
 
-  private[server] def initialize(zkClient: KafkaZkClient): Unit = {
+  private[server] def initialize(zkClient: Option[KafkaZkClient]): Unit = {
```

Comment:
It seems like this function does nothing if zkClient = None is passed.  Can we simply leave the function as-is and not call it in that case?